src/AdoNet/Orleans.Persistence.AdoNet/Options/AdoNetGrainStorageOptions.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Orleans.Persistence.AdoNet.Storage;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+
+namespace Orleans.Hosting
+{
+    public class AdoNetGrainStorageOptions
+    {
+        /// <summary>
+        /// Connection string for AdoNet storage
+        /// </summary>
+        public string ConnectionString { get; set; }
+
+        /// <summary>
+        /// Stage of silo lifecycle where storage should be initialized.  Storage must be initialzed prior to use.
+        /// </summary>
+        public int InitStage { get; set; } = DEFAULT_INIT_STAGE;
+        /// <summary>
+        /// Default init stage in silo lifecycle
+        /// </summary>
+        public const int DEFAULT_INIT_STAGE = SiloLifecycleStage.ApplicationServices;
+
+        /// <summary>
+        /// The default ADO.NET invariant used for storage if none is given. 
+        /// </summary>
+        public const string DEFAULT_ADONET_INVARIANTS = AdoNetInvariants.InvariantNameSqlServer;
+        /// <summary>
+        /// The invariant name for storage
+        /// </summary>
+        public string AdoInvariant { get; set; } = DEFAULT_ADONET_INVARIANTS;
+
+        #region json serialization related settings
+        /// <summary>
+        /// Whether storage string payload should be formatted in JSON.
+        /// </summary>
+        public bool UseJsonFormat { get; set; }
+        public bool UseFullAssemblyNames { get; set; }
+        public bool IndentJson { get; set; }
+        public TypeNameHandling? TypeNameHandling { get; set; }
+        #endregion
+        /// <summary>
+        /// Whether storage string payload should be formatted in Xml.
+        /// </summary>
+        public bool UseXmlFormat { get; set; }
+    }
+    public class AdoNetGrainStorageOptionsValidator : IConfigurationValidator
+    {
+        private readonly AdoNetGrainStorageOptions options;
+        private readonly string name;
+        public AdoNetGrainStorageOptionsValidator(AdoNetGrainStorageOptions configurationOptions, string name)
+        {
+            this.options = configurationOptions;
```

Comment:
It would definitely be cooler to check for parameters in the constructor. Even if it were "duplicate effort" that user is instructured optionally to use `ValidateConfiguration`.